### PR TITLE
Make SQL client general and add a CloudSQL connector

### PR DESCRIPTION
**Pull request by shifucun**
This is to prepare using SQL backend in the Cloud

### CODE REVIEW

File: internal/sqldb/sqlite/sqlite.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	);
+	`
+	_, err = sqlClient.Exec(tripleStatement)
+	if err != nil {
+		return nil, err
+	}
+
+	observationStatement := `
+	CREATE TABLE observations (
+		entity TEXT,
```
Comment (keyurva) [inline internal/sqldb/sqlite/sqlite.go:60]: Do we want this to be `entity` or `dcid`?

File: internal/server/data/import.go [hunk 2]
```diff
@@ -32,36 +30,30 @@ func Import(
 	in *pb.ImportRequest,
 	st *store.Store,
 	metadata *resource.Metadata,
-	openSql bool,
 ) (*resource.Cache, error) {
-	var err error
-	if err = writer.Write(
+	// First clear the tables.
+	err := sqldb.ClearTables(st.SQLClient, metadata.SQLitePath)
```
Comment (keyurva) [inline internal/server/data/import.go:35]: Is it worthwhile backing up vs deleting tables?

**Review by keyurva — COMMENTED** [review]
Thanks Bo. This generalization will help a lot.

File: internal/server/data/import.go (hunk 2 above)
Comment (shifucun) [reply internal/server/data/import.go:35]: I think eventually yes. I can imagine there is an endpoint for user to backup a table and have it in a "admin" UI. This is more of a prototype workflow for now.

File: internal/sqldb/sqlite/sqlite.go (hunk 1 above)
Comment (shifucun) [reply internal/sqldb/sqlite/sqlite.go:60]: should be dcid! I want to get e2e up running then clean this up.

**Review by keyurva — APPROVED** [review]

**Comment by shifucun** [comment]
thanks!

**Pull request merged**